xdsl/irdl/irdl.py

```diff
@@ -429,6 +429,115 @@ def infer(self, constraint_vars: dict[str, Attribute]) -> Attribute:
         return self.constr.infer(constraint_vars)
 
 
+class RangeConstraint(ABC):
+
+    @abstractmethod
+    def verify(
+        self, attrs: Sequence[Attribute | None], constraint_vars: dict[str, Attribute]
```

Comment:
I guess the `constraint_vars` is going to be changed later on when you create the new ranges variables?

Reply:
Indeed, probably by a more complete object encapsulating whatever state we want to add? At first just both variable types